File: go/tencent/list/reverse_list.go
package main

import "fmt"

func main() {
	res := new(ListNode)
	res.Val = 1
	res.Next = new(ListNode)
	res.Next.Val = 2
	res.Next.Next = new(ListNode)
	res.Next.Next.Val = 3
	res = reverseList(res)
	fmt.Println()
}

type ListNode struct {
	Val  int
	Next *ListNode
}

func reverseList(head *ListNode) *ListNode {
	var res *ListNode = nil
	for head != nil {
		next := head.Next
		head.Next = res
		res = head
		head = next
	}
	return res
}

File: java/leetcode-explore/highFrequency/list/ReverseList.java
package highFrequency.list;

public class ReverseList {
    public class ListNode {
        int val;
        ListNode next;

        ListNode() {
        }

        ListNode(int val) {
            this.val = val;
        }

        ListNode(int val, ListNode next) {
            this.val = val;
            this.next = next;
        }
    }

    public ListNode reverseList(ListNode head) {
        ListNode res = null;

        while (head != null) {
            ListNode next = head.next;
            head.next = res;
            res = head;
            head = next;
        }
        return res;
    }
}
